### PR TITLE
MBS-10774 extract DeviceProviderFactory creation

### PR DESCRIPTION
**Pull request by dsvoronin**
Preparation-refactoring

DeviceProviderFactory now created right in service,
next step will be direct creating kubernetesDeviceProvider there

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -124,6 +124,10 @@ publish_to_maven_local:
 publish_to_artifactory:
 	$(docker_command) ./gradlew $(project) $(log_level) $(params) publishToArtifactory -PprojectVersion=$(version) -Dorg.gradle.internal.publish.checksums.insecure=true
 
+stage_ui_tests:
```
Comment (eugene-krivobokov) [inline Makefile:127]: What is `stage`?

File: subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/internal/scheduling/TestsSchedulerFactory.kt [hunk 2]
```diff
@@ -15,56 +13,28 @@ import com.avito.instrumentation.internal.suite.filter.FilterInfoWriter
...
-import com.google.common.annotations.VisibleForTesting
+import com.avito.time.TimeProvider
 import com.google.gson.Gson
+import java.io.File
+import java.nio.file.Files
 
 internal interface TestsSchedulerFactory {
 
-    fun create(): TestsScheduler
+    fun create(devicesProviderFactory: DevicesProviderFactory): TestsScheduler
```
Comment (eugene-krivobokov) [inline subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/internal/scheduling/TestsSchedulerFactory.kt:23]: Can't we hide it in implementation class?

File: subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/service/TestRunParams.kt [hunk 3]
```diff
@@ -0,0 +1,20 @@
+package com.avito.instrumentation.service
+
+import com.avito.android.stats.SeriesName
+
+/**
+ * Parameters for a specific single test suite run
+ * Don't put global test runner parameters here
+ * "global" - means that it is available on a build start
```
Comment (eugene-krivobokov) [inline subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/service/TestRunParams.kt:8]: Where is this "global"? I don't see it in the class

File: subprojects/gradle/runner/device-provider/src/main/kotlin/com/avito/android/runner/devices/DevicesProviderFactory.kt [hunk 4]
```diff
@@ -1,13 +1,19 @@
...
         deviceType: DeviceType,
+        projectName: String,
         configurationName: String,
+        tempLogcatDir: File,
+        outputDir: File,
         logcatTags: Collection<String>,
-        kubernetesNamespace: String
+        kubernetesNamespace: String,
+        runnerPrefix: SeriesName
     ): DevicesProvider
```
Comment (eugene-krivobokov) [inline subprojects/gradle/runner/device-provider/src/main/kotlin/com/avito/android/runner/devices/DevicesProviderFactory.kt:18]: And here. These new parameters look like implementation details. I'd like to hide them from the public interface.

File: Makefile (hunk 1 above)
Comment (dsvoronin) [reply Makefile:127]: i wanted to express that it is ongoing changes, staged in terms of git, but not released yet

File: subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/internal/scheduling/TestsSchedulerFactory.kt (hunk 2 above)
Comment (dsvoronin) [reply subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/internal/scheduling/TestsSchedulerFactory.kt:23]: i'm afraid not in this moment of refactoring, i exposed it only to be able to provide it both for old action and new service ways

File: subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/service/TestRunParams.kt (hunk 3 above)
Comment (dsvoronin) [reply subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/service/TestRunParams.kt:8]: Global is parameters of service itself

File: subprojects/gradle/runner/device-provider/src/main/kotlin/com/avito/android/runner/devices/DevicesProviderFactory.kt (hunk 4 above)
Comment (dsvoronin) [reply subprojects/gradle/runner/device-provider/src/main/kotlin/com/avito/android/runner/devices/DevicesProviderFactory.kt:18]: true, i'll definitely hide it on later steps

**Review by eugene-krivobokov — APPROVED** [review]

**Pull request merged**